docs/reference/analysis/tokenfilters/apostrophe-tokenfilter.asciidoc

```diff
@@ -1,5 +1,84 @@
 [[analysis-apostrophe-tokenfilter]]
-=== Apostrophe Token Filter
+=== Apostrophe token filter
+++++
+<titleabbrev>Apostrophe</titleabbrev>
+++++
 
-The `apostrophe` token filter strips all characters after an apostrophe,
-including the apostrophe itself.
+Strips all characters after an apostrophe, including the apostrophe itself.
+
+[[analysis-apostrophe-tokenfilter-analyze-ex]]
+==== Example
+
+The following <<indices-analyze,analyze API>> request demonstrates how the
+apostrophe token filter works.
+
+[source,console]
+--------------------------------------------------
+GET /_analyze
+{
+  "tokenizer" : "standard",
+  "filter" : ["apostrophe"],
+  "text" : "Istanbul'a veya Istanbul'dan"
+}
+--------------------------------------------------
+
+The `apostrophe` token filter produces the following tokens:
+
+[source,text]
+---------------------------
+[ Istanbul, veya, Istanbul ]
+---------------------------
+
+/////////////////////
+[source,console-result]
+--------------------------------------------------
+{
+  "tokens" : [
+    {
+      "token" : "Istanbul",
+      "start_offset" : 0,
+      "end_offset" : 10,
+      "type" : "<ALPHANUM>",
+      "position" : 0
+    },
+    {
+      "token" : "veya",
+      "start_offset" : 11,
+      "end_offset" : 15,
+      "type" : "<ALPHANUM>",
+      "position" : 1
+    },
+    {
+      "token" : "Istanbul",
+      "start_offset" : 16,
+      "end_offset" : 28,
+      "type" : "<ALPHANUM>",
+      "position" : 2
+    }
+  ]
+}
+--------------------------------------------------
+/////////////////////
+
+[[analysis-apostrophe-tokenfilter-analyzer-ex]]
+==== Add to an analyzer
+
+The following <<indices-create-index,create index API>> request adds the
+`apostrophe` token filter to an analyzer.
+
+[source,console]
+--------------------------------------------------
+PUT /apostrophe_example
+{
+    "settings" : {
+        "analysis" : {
+            "analyzer" : {
+                "default" : {
```

Comment:
Small comment, maybe we don't want to always call the analyzer `default`. We could use a more specific name like `apostrophe` or `standard_apostrophe`.

Reply:
I changed the analyzer name to `standard_apostrophe` with 04a33a69ca37a8d687a6c23e2882f2b8c08e0ddd.

Reply:
To add to my comment: when experimenting with token filters, it seems more likely that the user will want to create a custom analyzer as opposed to overriding the `default` one. Maybe we could make a few small tweaks to clarify this, including using a specific analyzer name other than `default`and linking to the [custom analyzer](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-custom-analyzer.html) docs.

Reply:
Oops, I had a race condition with my comment. This looks good to me, we could also link to the custom analyzer docs if you think it's helpful.

Reply:
No worries! That makes sense to me.

I added a link to the custom analyzer docs with 701a349897747018e61a4238ad21a4b9c22e7372
and reworded this paragraph with a847eeb626e90fb1d6368f97c88a27d8d17c8150.